bio-primer-removal/main.py

```diff
@@ -0,0 +1,145 @@
+from typing import List, Tuple, Optional
+from pathlib import Path
+from zipfile import ZipFile, ZIP_DEFLATED
+
+import logging
+
+from coretex import Experiment, CustomDataset, CustomSample, folder_manager
+from coretex.project import initializeProject
+from coretex.bioinformatics import cutadaptTrim, isPairedEnd
+
+
+def forwardMetadata(sample: CustomSample, outputDataset: CustomDataset) -> None:
+    sample.unzip()
+
+    metadataZip = folder_manager.temp / "_metadata.zip"
+    with ZipFile(metadataZip, 'w', ZIP_DEFLATED) as archive:
+        for path in sample.path.iterdir():
+            archive.write(path, path.name)
+
+    if CustomSample.createCustomSample("_metadata", outputDataset.id, metadataZip) is None:
+        raise RuntimeError(">> [Microbiome analysis] Failed to forward metadata to the output dataset")
+
+
+def loadSingleEnd(sample: CustomSample) -> Path:
+    sample.unzip()
+
+    for filePath in sample.path.iterdir():
```

Comment:
There is no need for for-loop here, `sample.path.glob("*.fastq")` will do the trick. Also validate that the `glob` returns only a single path, if there is multiple then it should raise an error